pkg/source/reconciler/mtsource/kafkasource.go

```diff
@@ -158,6 +162,17 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, src *v1beta1.KafkaSource
 	return nil
 }
 
+func (r *Reconciler) FinalizeKind(ctx context.Context, src *v1beta1.KafkaSource) reconciler.Event {
+	if src.Status.Placement != nil {
+		src.Status.Placement = nil
+
+		// return an error to 1. update the status. 2. not clear the finalizer
+		return errors.New("placement list was not empty")
+	}
+
```

Comment:
Do we also need something to delete the MT receive adapters here?

Reply:
this is just removing the placements for 1 source only. The MT receiver adapter is still needed for the other sources

Reply:
Oh duh yes 🤦  thanks!